lib/auth/auth_with_roles.go

```diff
@@ -5481,6 +5481,95 @@ func (a *ServerWithRoles) DeleteAllSAMLIdPServiceProviders(ctx context.Context)
 	return a.authServer.DeleteAllSAMLIdPServiceProviders(ctx)
 }
 
+// CreatePlugin creates a new plugin instance.
+func (a *ServerWithRoles) CreatePlugin(ctx context.Context, req *proto.CreatePluginRequest) error {
+	if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbCreate); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := a.authServer.CreatePlugin(ctx, req); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// GetPlugin returns a plugin instance by name.
+func (a *ServerWithRoles) GetPlugin(ctx context.Context, name string, withSecrets bool) (types.Plugin, error) {
+	if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbReadNoSecrets); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if withSecrets {
+		if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbRead); err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	plugin, err := a.authServer.GetPlugin(ctx, name, withSecrets)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return plugin, nil
+}
+
+// GetPlugins returns all plugin instances.
+func (a *ServerWithRoles) GetPlugins(ctx context.Context, withSecrets bool) ([]types.Plugin, error) {
+	if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbList); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbReadNoSecrets); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if withSecrets {
+		if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbRead); err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
```

Comment:
```suggestion
	readVerb := types.VerbReadNoSecrets
	if withSecrets {
		readVerb = types.VerbRead
	}
	if err := a.action(apidefaults.Namespace, types.KindPlugin, types.VerbList, readVerb); err != nil {
		return nil, trace.Wrap(err)
	}
```

Would something like this work?